.github/actions/github-deployment-status-update/action.yml

```diff
@@ -0,0 +1,42 @@
+name: Update Deployment Status
+description: Update Deployment Status
+inputs:
+  github-token:
+    description: GitHub Token
+    required: true
+  deployment-id:
+    description: The Deployment Id
+    required: true
+  state:
+    description: >
+      The Deployment State to be set (error, failure, inactive, in_progress, queued, pending, success)
+      If empty, failure is used if failure is detected within the same job, success otherwise.
+
+runs:
+  using: "composite"
+  steps:
+    - name: Update deployment status
+      id: update
+      env:
+        GH_TOKEN: ${{ inputs.github-token }}
+        DEPLOYMENT_ID: ${{ inputs.deployment-id }}
+        STATE: ${{ inputs.state }}
+        REPO_URL: ${{ github.server_url }}/${{ github.repository }}
+        FAILURE: ${{ contains(steps.*.outcome, 'failure') }}
```

Comment:
this does not correctly allow to detect failure (job/step failure?) so not sure if this fallback can work after all...